hammer/vlsi/hammer_vlsi_impl.py

```diff
@@ -737,10 +745,9 @@ def specify_all_power_straps_by_tracks(self, layer_names: List[str], ground_net:
             blockage_spacing = coerce_to_grid(float(self._get_by_tracks_metal_setting("blockage_spacing", rail_layer_name)), rail_layer.grid_unit)
             # TODO does the CPF help this, or do we need to be more explicit about the bbox for each domain
             output.extend(self.specify_std_cell_power_straps(blockage_spacing, bbox, [ground_net] + power_nets))
-        # The layer to via down to
-        bottom_via_layer = rail_layer_name
+
         # The last layer we used
-        last = rail_layer
+        last = self.get_stackup().get_metal(bottom_via_layer)
```

Comment:
"rail" is not a valid name of a metal in any technology. Need to first do a if bottom_via_layer == "rail": <get the metal layer at index 1 (you can use `get_stackup().get_metal_by_index(1)`)>

Reply:
Thanks, the typo is actually triggering this.

Reply:
Oh, I see.